internal/pkg/template/templates/addons/aurora/rdws/serverlessv2.yml

```diff
@@ -0,0 +1,183 @@
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: !Sub
+        - Grants read access to the ${Secret} secret
+        - { Secret: !Ref {{logicalIDSafe .ClusterName}}AuroraSecret }
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: SecretActions
+            Effect: Allow
+            Action:
+              - 'secretsmanager:GetSecretValue'
+            Resource:
+              - !Ref {{logicalIDSafe .ClusterName}}AuroraSecret
+
+  {{- if .ParameterGroup}}
+  # {{logicalIDSafe .ClusterName}}DBClusterParameterGroup:
+  #   Type: 'AWS::RDS::DBClusterParameterGroup'
+  #   Properties:
+  #     Description: !Ref 'AWS::StackName'
+  #     Family: 'aurora-mysql8.0'
+  #     Parameters:
+  #       character_set_client: 'utf8'
+  {{- else}}
+  {{logicalIDSafe .ClusterName}}DBClusterParameterGroup:
+    Metadata:
+      'aws:copilot:description': 'A DB parameter group for engine configuration values'
+    Type: 'AWS::RDS::DBClusterParameterGroup'
+    Properties:
+      Description: !Ref 'AWS::StackName'
+      {{- if eq .Engine "MySQL"}}
+      Family: 'aurora-mysql8.0'
+      Parameters:
+        character_set_client: 'utf8'
+      {{- else}}
+      Family: 'aurora-postgresql14'
+      Parameters:
+        client_encoding: 'UTF8'
+      {{- end}}
+  {{- end}}
+  {{logicalIDSafe .ClusterName}}DBCluster:
+    Metadata:
+      'aws:copilot:description': 'The {{logicalIDSafe .ClusterName}} Aurora Serverless v2 database cluster'
+    Type: 'AWS::RDS::DBCluster'
+    Properties:
+      MasterUsername:
+        !Join [ "",  [ {{`'{{resolve:secretsmanager:'`}}, !Ref {{logicalIDSafe .ClusterName}}AuroraSecret, ":SecretString:username}}" ]]
+      MasterUserPassword:
+        !Join [ "",  [ {{`'{{resolve:secretsmanager:'`}}, !Ref {{logicalIDSafe .ClusterName}}AuroraSecret, ":SecretString:password}}" ]]
+      DatabaseName: !Ref {{logicalIDSafe .ClusterName}}DBName
+      {{- if eq .Engine "MySQL"}}
+      Engine: 'aurora-mysql'
+      EngineVersion: '8.0.mysql_aurora.3.02.0'
+      {{- else}}
+      Engine: 'aurora-postgresql'
+      EngineVersion: '14.4'
+      {{- end}}
+      DBClusterParameterGroupName: {{- if .ParameterGroup}} {{.ParameterGroup}} {{- else}} !Ref {{logicalIDSafe .ClusterName}}DBClusterParameterGroup {{- end}}
+      DBSubnetGroupName: !Ref {{logicalIDSafe .ClusterName}}DBSubnetGroup
+      VpcSecurityGroupIds:
+        - !Ref {{logicalIDSafe .ClusterName}}DBClusterSecurityGroup
+      ServerlessV2ScalingConfiguration:
+        # Replace "All" below with "!Ref Env" to set different autoscaling limits per environment.
+        MinCapacity: !FindInMap [{{logicalIDSafe .ClusterName}}EnvScalingConfigurationMap, All, DBMinCapacity]
+        MaxCapacity: !FindInMap [{{logicalIDSafe .ClusterName}}EnvScalingConfigurationMap, All, DBMaxCapacity]
+  {{logicalIDSafe .ClusterName}}DBWriterInstance:
```

Comment:
Oh very interesting, can you explain to be why an explicit writer instance is needed?
We didn't set one for aurora serverless v1 so I'm curious why v2 is different?

Reply:
We had an offline discussion and I'm sharing it here for visibility:

> Does RDS automatically create a writer instance for that cluster?

For v2, it doesn't. If we have only the `DBCluster` resource in the template, only a cluster is created - no writer instance.

> We didn't set one for aurora serverless v1 so I'm curious why v2 is different?

v1 contains one and only one read/write instance. When we create a `DBCluster`, that one and only one read/write instance also gets automatically created. This is different from v2 and is why we didn't need to explicitly create an instance before.